File: app/src/main/java/com/wind/androidlearn/network/converter/DecodeRequestBodyConverter.java
package com.wind.androidlearn.network.converter;

import com.google.gson.Gson;
import com.google.gson.TypeAdapter;
import com.wind.androidlearn.bassis.Utils.LogUtil;
import com.wind.androidlearn.network.utils.GsonUtil;
import com.wyt.zdf.myapplication.BuildConfig;


import java.io.IOException;
import java.util.Map;

import okhttp3.MediaType;
import okhttp3.RequestBody;
import retrofit2.Converter;


/**
 * 加密DecodeRequestBodyConverter
 */
public final class DecodeRequestBodyConverter<T> implements Converter<T, RequestBody> {

    private static final MediaType MEDIA_TYPE = MediaType.parse("application/x-www-form-urlencoded; charset=UTF-8");

    DecodeRequestBodyConverter(Gson gson,TypeAdapter<T> adapter){

    }
    @Override
    public RequestBody convert(final T value) throws IOException {
        //获得value的json
        String jsonStirng = GsonUtil.GsonString(value);
        LogUtil.d("ApiFactory","加密前的请求参数 = " + jsonStirng);
        //参数转化成Map
        Map<String,String> map = GsonUtil.GsonToMaps(jsonStirng);
        //获得加密字符串
        String json = Base64Utils.getEncodedPostString(map, BuildConfig.REQUEST_KEY);
        return RequestBody.create(MEDIA_TYPE,json);
    }
}

File: common/src/main/java/com/wyt/common/network/converter/DecodeRequestBodyConverter.kt
package com.wyt.common.network.converter

import com.wyt.common.config.Constants
import com.wyt.common.config.Constants.Http.MEDIA_TYPE
import com.wyt.common.utils.GsonUtil
import com.wyt.common.utils.Log
import com.wyt.common.utils.RobotRequestUtil
import okhttp3.RequestBody
import retrofit2.Converter

/**
 * 机器人后台的加密DecodeRequestBodyConverter
 */
class DecodeRequestBodyConverter<T> : Converter<T, RequestBody> {
    override fun convert(value: T): RequestBody {
        //获得value的json
        val jsonStirng = GsonUtil.gsonString(value)
        Log.d("ApiFactory加密前的请求参数 = $jsonStirng")
        //参数转化成Map
        val map =
            GsonUtil.gson2Map(jsonStirng)
        //获得加密字符串
        val json = RobotRequestUtil.getEncodedPostString(
            map,
            Constants.Http.ROBOT_REQUEST_KEY
        )
        Log.d("ApiFactory加密后的请求参数 = $json")
        return RequestBody.create(
            MEDIA_TYPE,
            json
        )
    }
}